emmet-core/emmet/core/search.py

```diff
@@ -261,3 +261,133 @@ class SearchDoc(PropertyDoc):
     # Theoretical
 
     theoretical: bool = Field(True, description="Whether the material is theoretical.")
+
+    @classmethod
+    def from_docs(cls, material_id: MPID, **docs: Dict[str, Dict]):
+        """Converts a bunch of search docs into a SearchDoc"""
+        doc = _copy_from_doc(docs)
+
+        # Reshape document for various sub-sections
+        # Electronic Structure + Bandstructure + DOS
+        if "bandstructure" in doc:
+            doc["has_props"].append("bandstructure")
+        if "dos" in doc:
+            doc["has_props"].append("dos")
+        if "calc_id" in doc:
```

Comment:
`calc_id` should be switched back to `task_id` to match `ElectronicStructureBaseData`